### DISCUSSION TITLE
TypeORM adapter - how to change the default database table names

### DISCUSSION THREAD

**Discussion by oldo** [nextauthjs/next-auth#2050] [Help] [answered]
Hi there,
I'd like to change the table names for some of the tables that are created using the default database bootstrapping SQL query. For example, I'd like to use the table name `accounts` in my app for actual business accounts, not for oauth accounts.
Would someone be able to point me in the right direction to where I can modify the TypeORM adapter to use different table names?
Thanks! :-)

**Comment by oldo** [comment, marked as answer]
I've been able to figure this out and turned out to be fairly simple.
Follow this tutorial for defining custom models: https://next-auth.js.org/tutorials/typeorm-custom-models
And TypeORM defines the table name by the name of the class that defines the model. So for example, to change the table name for `accounts` to `user_oauth_accounts`, just change the class name as follows:

```js
import Adapters from 'next-auth/adapters';

// the class name sets the table that is used for this model
export class UserOauthAccount extends Adapters.TypeORM.Models.Account.model {
  // You can extend the options in a model but you should not remove the base
  // properties or change the order of the built-in options on the constructor
  // see: https://github.com/nextauthjs/adapters/tree/canary/packages/typeorm-legacy/src/models
  // for the appropriate constructor args
  constructor(
    userId,
    providerId,
    providerType,
    providerAccountId,
    refreshToken,
    accessToken,
    accessTokenExpires
  ) {
    super(
      userId,
      providerId,
      providerType,
      providerAccountId,
      refreshToken,
      accessToken,
      accessTokenExpires
    );
  }
}

export const UserOauthAccountSchema = {
  name: 'UserOauthAccount',
  target: UserOauthAccount,
  columns: {
    ...Adapters.TypeORM.Models.Account.schema.columns,
  },
};
```

The class name is transformed by the pluralized snake case function: https://github.com/nextauthjs/adapters/blob/canary/packages/typeorm-legacy/src/lib/naming-strategies.js#L5

**Reply by balazsorban44** [reply]
glad you found it out!

**Reply by oldo** [reply]
Thank you, and apologies again for the duplicate post!
